back-end/app/Http/Controllers/ContractController.php

```diff
@@ -87,4 +88,127 @@ public function store(Request $request)
+     *     ),
+     *     @OA\Response(
+     *         response=401,
+     *         description="Unauthorized"
+     *     ),
+     *     @OA\Response(
+     *         response=403,
+     *         description="Forbidden - User is not an admin"
+     *     ),
+     *     @OA\Response(
+     *         response=500,
+     *         description="Internal Server Error"
+     *     ),
+     *     security={{"bearerAuth":{}}},
+     * )
+     */
+    public function show()
+    {
+        $existsInUserContract = Contracts::join('user_contract', 'contracts.ContractID', '=', 'user_contract.ContractID')
+            ->select('contracts.*')
+            ->get();
+
+        $notExistsInUserContract = Contracts::whereNotIn('ContractID', function ($query) {
+            $query->select('ContractID')->from('user_contract');
+        })->get();
+
+        return response()->json([
+            'existsInUserContract' => $existsInUserContract,
+            'notExistsInUserContract' => $notExistsInUserContract,
+        ], 200);
+    }
+
+    /**
+     * @OA\Delete(
+     *     path="/api/contract/delete",
+     *     tags={"Contracts"},
+     *     summary="Delete a contract",
+     *     description="Delete a contract by its ID if it does not have an assigned user in the user_contract table.",
+     *     @OA\RequestBody(
+     *         required=true,
+     *         @OA\JsonContent(
+     *             @OA\Property(property="id", type="integer", description="The ID of the contract to be deleted", example=1)
+     *         )
+     *     ),
+     *     @OA\Response(
+     *         response=200,
+     *         description="Contract deleted",
+     *         @OA\JsonContent(
+     *             @OA\Property(property="message", type="string", example="Contract deleted")
+     *         )
+     *     ),
+     *     @OA\Response(
+     *         response=403,
+     *         description="Forbidden - The contract has an assigned user",
+     *         @OA\JsonContent(
+     *             @OA\Property(property="message", type="string", example="The contract has a assigned user")
+     *         )
+     *     ),
+     *     @OA\Response(
+     *         response=422,
+     *         description="Validation error - Missing or invalid input",
+     *         @OA\JsonContent(
+     *             @OA\Property(property="message", type="string", example="The id field is required.")
+     *         )
+     *     ),
+     *     @OA\Response(
+     *         response=401,
+     *         description="Unauthorized - Invalid or missing token"
+     *     ),
+     *     @OA\Response(
+     *         response=500,
+     *         description="Internal Server Error"
+     *     ),
+     *     security={{"bearerAuth":{}}},
+     * )
+     */
+    public function delete(Request $request)
+    {
+        $request->validate([
+            'id' => 'required|integer|exists:contracts,ContractID',
+        ]);
+
+        if (UserContract::where('ContractID', $request->id)->exists()) {
```

Comment:
Het lijkt mij essensieel om hier ook te kijken of het contract in gebruik is door andere gebruikers. Als dit namelijk wel is dan gaan de relaties kapot.